examples/fastly/src/index.js

```diff
@@ -0,0 +1,79 @@
+/// <reference types="@fastly/js-compute" />
+import { connect } from '@planetscale/database'
+
+
+addEventListener("fetch", (event) => event.respondWith(app(event)))
+
+fastly.defaultBackend = "psdb"
+const CURRENT_YEAR = 2022
+/**
+ *
+ * @param {FetchEvent} event
+ * @returns {Promise<Response>}
+ */
+async function app(event) {
+  const secrets = new Dictionary('secrets')
+  const url = new URL(event.request.url)
+  const pathname = url.pathname
+
+  const config = {
+    host: secrets.get('PSCALE_HOST'),
+    username: secrets.get('PSCALE_USERNAME'),
+    password: secrets.get('PSCALE_PASSWORD')
+  }
+
+  if (pathname === '/data.json') {
+    const conn = connect(config)
+    const [resultsData, raceData, standingsData] = await Promise.all([
+      conn.execute(
+        'SELECT constructor_standings.round, race_name, teamId, name, nationality, url, position, points FROM constructor_standings JOIN constructor_teams ON constructor_standings.teamId = constructor_teams.id JOIN constructor_races on (constructor_standings.season = constructor_races.season AND constructor_standings.round = constructor_races.round) where constructor_standings.season = ? order by constructor_standings.round',
+        [CURRENT_YEAR]
+      ),
+      conn.execute('SELECT * FROM constructor_races WHERE season = ?', [CURRENT_YEAR]),
+      conn.execute('SELECT constructor_standings.season, teamId, name, position, points, wins FROM constructor_standings JOIN constructor_teams on (constructor_standings.teamId = constructor_teams.id) WHERE season = ? and round = (select max(round) from constructor_standings where season = ?) order by position', [CURRENT_YEAR, CURRENT_YEAR])
+    ])
+
+    const data = { races: raceData.rows, results: resultsData.rows, standings: standingsData.rows }
+    const json = JSON.stringify(data)
+
+    return new Response(json, {
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8',
+        'Access-Control-Allow-Origin': '*'
+      }
+    })
+  } else {
+    return new Response('Not Found', {
+      status: 404,
+      headers: { 'Content-Type': 'text/html' }
+    })
+  }
+}
+
+/**
+ * Polyfill for bota (Binary to ASCII) function
+ * @param {String} binary
+ * @returns {String}
+ */
+const ascii = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/='
+globalThis.btoa = function btoa(binary) {
```

Comment:
@JakeChampion Did you end up still needing this? Or did this get added into the Fastly runtime? Because if it's still needed, you'd need to stub in `atob` as well, but I'm guessing if you've gotten this far, you don't need this.

Reply:
It's not yet in the Fastly runtime, it should land in either the next release or the one after that 
I'll add a polyfill for `atob` as well and when Fastly's JS runtime supports both functions I can update the example to use the latest runtime and remove the polyfills 👍

Reply:
This is the pull-request which implements btoa and atob in the Fastly runtime -> https://github.com/fastly/js-compute-runtime/pull/227